Add claim info to project details

TethysPlex opened the pull request:
closes #38

TethysPlex committed:
Add claim info to project details

TethysPlex committed:
Merge pull request #3 from Shinplex/codex/add领取字段到项目api

Add claim info for project API

TethysPlex committed:
docs: update api docs

yyg-max commented on internal/apps/project/routers.go:
if err == nil {}这判断可以不需要看着很怪

U8F69 commented on internal/apps/project/models.go:
```diff
@@ -191,3 +191,11 @@ func (p *ProjectItem) Exact(tx *gorm.DB, id uint64) error {
 	return nil
 }
+
+func (p *Project) GetReceivedItem(ctx context.Context, userID uint64) (*ProjectItem, error) {
```
GetReceivedItem 这个如果只是用来判定是否领取的话，比如直接返回 bool, error，而无需返回整个 ProjectItem 对象

TethysPlex committed:
Merge branch 'chenyme:dev' into dev

TheNeuroplexus committed:
fix(project): handle not found error correctly when getting received item

- GetReceivedItem returns nil without error if no record is found
- Update caller to check for nil item instead of error for not found case

TheNeuroplexus committed:
style(project): remove unused errors import

TheNeuroplexus committed:
refactor(project): clean import statements and remove duplicates

TheNeuroplexus committed:
fix(project): 修正导入语句中的拼写错误
